src/multi-select/header.tsx

```diff
@@ -7,16 +7,23 @@ export const DropdownHeader = () => {
 
   const noneSelected = value.length === 0;
   const allSelected = value.length === options.length;
-  const customText = valueRenderer && valueRenderer(value, options);
+  const selectedText = valueRenderer
+    ? valueRenderer(value, options)
+    : "Text Undefined";
 
-  const getSelectedText = () => value.map((s) => s.label).join(", ");
-
-  return noneSelected ? (
-    <span className="gray">{customText || t("selectSomeItems")}</span>
-  ) : (
-    <span>
-      {customText ||
-        (allSelected ? t("allItemsAreSelected") : getSelectedText())}
-    </span>
+  const getSelectedText = useMemo(
+    () => () => value.map((s) => s.label).join(", "),
+    [value]
```

Comment:
using the useMemo hook to memoize the getSelectedText function, to avoid recomputing it on every render.

Reply:
Why are you overcomplicating this by returning a function from `useMemo` ? 

Could be simply a string returned from useMemo with no invocation